packages/api/source/controllers/MembershipController.ts

```diff
@@ -0,0 +1,157 @@
+import type { Document } from "mongoose";
+import joi from "joi";
+import {
+    MembershipModel,
+    UserModel,
+    constants,
+    BadRequestError,
+    NotFoundError,
+    OrganizationModel,
+    ExternalMembership,
+    runAsTransaction,
+} from "@hypertool/common";
+import jwt from "jsonwebtoken";
+
+import { invitationTemplate, sendEmail } from "../utils";
+
+const { INVITATION_JWT_SIGNATURE } = process.env;
+
+const createSchema = joi.object({
+    emailAddress: joi.string().email().required(),
+    organizationId: joi.string().regex(constants.identifierPattern),
+    inviterId: joi.string().regex(constants.identifierPattern),
+});
+
+const toExternal = (membership: any): ExternalMembership => {
+    const {
+        id,
+        _id,
+        member,
+        inviter,
+        division,
+        type,
+        status,
+        createdAt,
+        updatedAt,
+    } = membership;
+    return {
+        id: id || _id.toString(),
+        member,
+        inviter,
+        division,
+        type,
+        status,
+        createdAt,
+        updatedAt,
+    };
+};
+
+const create = async (context, attributes): Promise<ExternalMembership> => {
+    const { error, value } = createSchema.validate(attributes, {
+        stripUnknown: true,
+    });
+
+    if (error) {
+        throw new BadRequestError(error.message);
+    }
+
+    const { emailAddress, organizationId, inviterId } = value;
+    let member = await UserModel.findOne({ emailAddress });
+
+    if (!member) {
+        member = new UserModel({
+            firstName: "<unavailable>",
+            lastName: "<unavailable>",
+            description: "<unavailable>",
+            gender: undefined,
+            countryCode: undefined,
+            emailAddress: emailAddress,
+            emailVerified: true,
+            status: "invited",
+            role: "viewer",
+        });
+
+        await member.save();
+    }
+
+    let membership = await MembershipModel.findOne({
+        member: member.id,
+        organizationId,
+    });
+
+    if (!membership) {
+        membership = new MembershipModel({
+            member: member.id,
+            inviter: inviterId,
+            division: organizationId,
+            type: "organization",
+            status: "invited",
+        });
+        await membership.save();
+    }
+
+    if (membership.status === "invited") {
```

Comment:
I think you were supposed to throw an exception when the status is anything other than `invited`. Please refer the ticket.